### PR TITLE
DataFusion Substrait blog post

### PR DESCRIPTION
**Pull request by andygrove**

### CODE REVIEW

**Comment by andygrove** [comment]
@jdye64 @nseekhao fyi

**Review by avantgardnerio — APPROVED** [review]

**Comment by kevingurney** [comment]
@alamb - it seems like the Webpack issues linked below may be causing the build failure:

1. https://github.com/webpack/webpack/issues/14532#issuecomment-1434095919
2. https://github.com/webpack/webpack/issues/13572
3. https://github.com/webpack/webpack/pull/14306

My high level understanding is that in Node 18 (the build output shows Node.js v18.14.1 is being used), the `md4` hashing algorithm is deprecated and the version of Webpack used by `apache/arrow-site` (v5.21.2) seems to default to using `md4`.

[Webpack v5.61.0](https://github.com/webpack/webpack/releases/tag/v5.61.0) added a WASM `md4` implementation as a fallback. However, the advice in [1.](https://github.com/webpack/webpack/issues/14532#issuecomment-1434095919) recommends setting `output.hasFunction` in the Webpack config to use an alternative hashing algorithm instead. Specifically, it recommends using `xxhash64` (which is planned to be the default hashing algorithm when Webpack 6 is released).

So, to summarize, it seems that a combination of upgrading to the latest version of Webpack and switching over to `xxhash64` may resolve this issue.

**Comment by kevingurney** [comment]
After some more investigation, I discovered that the [default Node.js version was switched to 18 for the `ubuntu-latest` GitHub Actions runner image on February 13th](https://github.com/actions/runner-images/issues/7002 ). This seems to explain why this suddenly started failing a few weeks ago.

**Comment by kevingurney** [comment]
I've captured this issue as a Bug with the Component set to `[Website]` in [#34379](https://github.com/apache/arrow/issues/34379) in the `apache/arrow` project.

**Comment by avantgardnerio** [comment]
> After some more investigation, I discovered that...

Personally, I would recommend that we switch away from the proprietary, arbitrarily changing, and non-locally debuggable `ubuntu-latest` and to the official locally-runnable docker image `ubuntu:latest` where possible as we did here:

https://github.com/apache/arrow-ballista/blob/b61cfbf54705f4cbfcbc7103f87509e49cd01fda/.github/workflows/rust.yml#L79

**Comment by kevingurney** [comment]
Thanks @avantgardnerio! Agreed - this seems like a good solution. It would be great to not have to worry about things changing suddenly in `ubuntu-latest`.

**Comment by alamb** [comment]
> Thanks @avantgardnerio! Agreed - this seems like a good solution. It would be great to not have to worry about things changing suddenly in ubuntu-latest. I've added this to the list of potential workarounds in https://github.com/apache/arrow/issues/34379.

Awesome -- could someone make a PR to this repo?

**Comment by kevingurney** [comment]
@alamb - yes, I just assigned the issue to myself and will make a PR

**Comment by kevingurney** [comment]
Update: I've opened pull request #326 to address the build issues.

**Comment by alamb** [comment]
The doc build has been fixed -- do we still want to publish this content?

**Comment by ianmcook** [comment]
Closing this PR due to inactivity. Please feel free to reopen if there are updates.
